runtime/v1/linux/bundle.go

```diff
@@ -127,10 +128,32 @@ func (b *bundle) Delete() error {
 	return errors.Wrapf(err, "Failed to remove both bundle and workdir locations: %v", err2)
 }
 
-func (b *bundle) shimAddress(namespace string) string {
+func (b *bundle) legacyShimAddress(namespace string) string {
 	return filepath.Join(string(filepath.Separator), "containerd-shim", namespace, b.id, "shim.sock")
 }
 
+func (b *bundle) shimAddress(namespace string) string {
+	d := sha256.Sum256([]byte(filepath.Join(namespace, b.id)))
+	return filepath.Join(string(filepath.Separator), "containerd-shim", fmt.Sprintf("%x.sock", d))
+}
+
+func (b *bundle) loadAddress() (string, error) {
+	addressPath := filepath.Join(b.path, "address")
+	data, err := ioutil.ReadFile(addressPath)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+func (b *bundle) decideShimAddress(namespace string) string {
+	address, err := b.loadAddress()
+	if err != nil {
+		return address
```

Comment:
I think we should return legacyone if the err is not nil(notfound), right?

Reply:
This logic looks wrong here.  We should handle an `os.IsNotExist` errror and then call `legacyShimAddress`

Reply:
Just asking，if `err == IsNotExist`, the returning `address` would be empty, it seems we pass a potential err to `Connect`

Reply:
I think it is fine to ignore the error here.

Since there is no `ctx` pass-through var, it should use `log.G(context.TODO())`. I don't know it is good way to add it. 

ping @AkihiroSuda @crosbymichael

Reply:
this handling still looks wrong.  if we get an error here then you should be returning the legacy address right?!